### ISSUE TITLE
Support for fancy indexing

### ISSUE THREAD

**Issue by JeanKossaifi** [apache/incubator-mxnet#8084]
Fancy indexing --at least, being able to index one dimension of an NDArray with a list of indices-- is a vital functionality needed for a lot of applications. Without it one has to loop through dimensions and elements which is prohibitively slow.  PR #7910 addresses this partly.

However, syntaxes such as 
```python
tensor[:, [1, 2, 3]]
```
are still not supported.

Given the importance of this functionality, at least an efficient workaround is needed until this can be fixed.
 
One such workaround would be to index the dimensions to keep (`:`) with a list containing all indices as follows:
```python
a  =  mx.nd.array(np.random.random((20, 20, 20)))
a[tuple(list(range(i)) for i in a.shape)]
```
While this works as expected, this fails:
```python
a[list(range(a.shape[0])), list(range(a.shape[1])), [1, 3, 4]]
```

Another possibility would be to extend `mxnet.ndarray.take` to support several axis, or at least allow to support axis other than `0`.

**Comment by JeanKossaifi** [comment]
Given the importance of the feature and the current fix used (a warning message) this should maybe be classified as a bug?

**Comment by dai-dao** [comment]
Hi, Just want to ask if there's any advancements for this feature? Thank you

**Comment by JeanKossaifi** [comment]
Addressed by #8246.

**Comment by meanmee** [comment]
could this new feature support the operation like np.where()？
